enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/log/cache/InFlightCacheFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.core.consensus.log.cache;
+
+import org.neo4j.causalclustering.core.CausalClusteringSettings;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.monitoring.Monitors;
+
+import static org.neo4j.causalclustering.core.CausalClusteringSettings.in_flight_cache_max_bytes;
+import static org.neo4j.causalclustering.core.CausalClusteringSettings.in_flight_cache_max_entries;
+
+public class InFlightCacheFactory
+{
+    public static InFlightCache create( Config config, Monitors monitors )
+    {
+        Type type = config.get( CausalClusteringSettings.in_flight_cache_type );
+
+        switch ( type )
```

Comment:
This logic could be included in the enum as an overridden method

Reply:
Good idea, will help me get rid of that pesky default case.